File: scripts/io-bound/io-bound.pl
#!/usr/bin/env perl

use v5.38;
use threads;
use LWP::Simple;
use Time::HiRes qw(time);

sub io_worker($id) {
    my $url = "https://www.google.com";
    my $content = get($url);
    say "Thread $id finished.";
}

my $start_time = time();
my @threads = map { threads->create(\&io_worker, $_) } (1..9);
$_->join for @threads;
my $end_time = time();
printf("Total time taken: %.2f seconds\n", $end_time - $start_time);

File: scripts/io-bound/io-bound.py
#!/usr/bin/env python3

import time
import requests
import threading

def io_worker(id):
    url = "https://www.google.com"
    response = requests.get(url)
    print(f"Thread {id} finished.")

start_time = time.time()
threads = []
for i in range(9):
    thread = threading.Thread(target=io_worker, args=(i + 1,))
    threads.append(thread)
    thread.start()
for thread in threads:
    thread.join()
end_time = time.time()
print(f"Total time taken: {end_time - start_time:.2f} seconds")
